server/neptune/workflows/internal/activities/execute.go

```diff
@@ -1,12 +1,90 @@
 package activities
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
 
-type executeCommandActivities struct{}
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime/cache"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/steps"
+)
+
+type executeCommandActivities struct {
+	versionCache cache.ExecutionVersionCache
+
+	DefaultTFVersion *version.Version
+
+	// TerraformBinDir is the directory where Atlantis downloads Terraform binaries.
+	TerraformBinDir string
+}
+
+type ExecuteCommandResponse struct {
+	Output string
+	Error  error
+}
 
 type ExecuteCommandRequest struct {
+	Step          steps.Step
+	CustomEnvVars map[string]string
+	Envs          map[string]string
+
+	Path string
+	// TerraformVersion is the version of terraform we should use when executing
+	// commands for this project. This can be set to nil in which case we will
+	// use the default Atlantis terraform version.
+	TerraformVersion *version.Version
 }
 
-func (t *terraformActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) error {
-	return nil
+func (t *executeCommandActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) ExecuteCommandResponse {
+
+	terraformVersion := t.DefaultTFVersion
+	if request.TerraformVersion != nil {
+		terraformVersion = request.TerraformVersion
+	}
+
+	// Ensure version exists
+	_, err := t.versionCache.Get(terraformVersion)
+	if err != nil {
+		return ExecuteCommandResponse{
+			Output: "",
+			Error:  errors.Wrapf(err, "getting version %s", request.TerraformVersion),
+		}
+	}
+
+	terraformEnvVars := map[string]string{
+		"ATLANTIS_TERRAFORM_VERSION": terraformVersion.String(),
+		"PATH":                       fmt.Sprintf("%s:%s", os.Getenv("PATH"), t.TerraformBinDir),
+	}
+
+	cmd := exec.Command("sh", "-c", request.Step.RunCommand) // #nosec
+	cmd.Dir = request.Path
+
+	baseEnvVars := os.Environ()
+
+	finalEnvVars := baseEnvVars
+	for key, val := range request.CustomEnvVars {
+		finalEnvVars = append(finalEnvVars, fmt.Sprintf("%s=%s", key, val))
+	}
+	for key, val := range request.Envs {
+		finalEnvVars = append(finalEnvVars, fmt.Sprintf("%s=%s", key, val))
+	}
+	for key, val := range terraformEnvVars {
+		finalEnvVars = append(finalEnvVars, fmt.Sprintf("%s=%s", key, val))
+	}
+	cmd.Env = finalEnvVars
+	out, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return ExecuteCommandResponse{
+			Output: "",
+			Error:  fmt.Errorf("%s: running %q in %q: \n%s", err, request.Step.RunCommand, request.Path, out),
+		}
+	}
+	return ExecuteCommandResponse{
+		Output: string(out),
```

Comment:
We shouldn't return the output from the activity, Activity inputs and outputs are serialized over the wire and as such have size limitations.  We should think about how to solve this as we'll need to do this with the Terraform steps as well.  One option is storing this data somewhere and passing around a pointer to it basically. And then whenever we write that data to our GH checks we just pull from that location.

Probably would be good to spec something out real quick on how we could do this.

Reply:
Hmm yeah, I didn't think much about that part since we only run custom scripts and env steps in this activity so the size of the output should not be an issue (unless the script does something huge) compared the the plan and apply operations. 

To keep this PR short, I'll handle this issue and streaming logs for the operation in a follow up task which we can pull in to this sprint. One simple approach would be have a directory structure inside the workspace we're using to write output to files which can be read outside the worker. What do you think?

Reply:
like i said I think it requires some analysis on tradeoffs which should probably not be in a PR comment. I think its fine to go with this for now pls create a ticket to track this, I assumed this was also being called for all tf operations as well.

Reply:
Created a ticket for this in our backlog: https://jira.lyft.net/browse/WENGINES-4727 We can probably bring it into the next sprint once we have an e2e terraform workflow running